searxng/searxng
# mudularize & document searx.results and fix #1348

## What does this PR do?

Fix issues with hostname_replace where URLs in answers and infobox not handled.

In preparation for the bugfix, the first commit mudularize & document searx.results.

```
- [mod] mudularize & document searx.results
- [fix] hostname_replace: modify URLs in the infobox
```

## Why is this change important?

Modularize code, add documentation and fix an issue.



## How to test this PR locally?

First check the documentation `make docs.clean docs.live` and visit pages:

- dev/engine_overview.html (read "Making a Response")
- src/searx.results.html

Second test hostname_replace plugin by redirecting youtube and twitter:

```yaml
hostname_replace:
  # youtube --> Invidious

  '(.*\.)?youtube-nocookie\.com': 'invidio.xamh.de'
  '(.*\.)?youtube\.com$': 'invidio.xamh.de'

  # twitter --> nitter

  '(www\.)?twitter\.com$': 'nitter.net'

  # test "!cc 1 EUR to dollar" --> link should be removed

  '(.*\.)?duckduckgo\.com$': false
```

make a search, in the infobox the links should be redirected.

## Related issues

Closes: https://github.com/searxng/searxng/issues/1348
Supersede: #1356
